src/libraries/System.IO.MemoryMappedFiles/src/System/IO/MemoryMappedFiles/MemoryMappedFile.Unix.cs

```diff
@@ -162,11 +162,14 @@ private static FileAccess TranslateProtectionsToFileAccess(Interop.Sys.MemoryMap
 
         private static SafeFileHandle CreateSharedBackingObject(Interop.Sys.MemoryMappedProtections protections, long capacity, HandleInheritability inheritability)
         {
-            return CreateSharedBackingObjectUsingMemory(protections, capacity, inheritability)
-                ?? CreateSharedBackingObjectUsingFile(protections, capacity, inheritability);
+            SafeFileHandle? handle = Interop.Sys.IsMemfdSupported ?
+                CreateSharedBackingObjectUsingMemoryMemfdCreate(protections, capacity, inheritability) :
+                CreateSharedBackingObjectUsingMemoryShmOpen(protections, capacity, inheritability);
+
+            return handle ?? CreateSharedBackingObjectUsingFile(protections, capacity, inheritability);
```

Comment:
This fallback path is now only relevant when using CreateSharedBackingObjectUsingMemoryShmOpen, right?

Reply:
Yup, you are right. Fixed.